src/server/models/cacheable_queries/question-response.js

```diff
@@ -0,0 +1,53 @@
+import { r } from '../../models'
+
+const responseCacheKey = (campaignContactId) => (
+  `${process.env.CACHE_PREFIX || ''}response-${campaignContactId}`
+)
+
+const questionResponseCache = {
+  query: async (campaignContactId, minimalObj) => {
+    // For now, minimalObj is always being invoked as true in
+    // server/api/campaign-contact
+    if (r.redis && minimalObj) {
+      const cachedResponse = await r.redis.hgetallAsync(
+        responseCacheKey(campaignContactId)
+      )
+      if (cachedResponse) {
+        const formattedResponse = Object.keys(cachedResponse).reduce((acc, key) => {
+          acc.push({ interaction_step_id: key, value: cachedResponse[key] })
+          return acc
+        }, [])
+        return formattedResponse
+      }
+    }
+    return await r.knex('question_response')
+      .where('question_response.campaign_contact_id', campaignContactId)
+      .select('value', 'interaction_step_id')
+  },
+  clearQuery: async (campaignContactId) => {
+    if (r.redis) {
+      await r.redis.delAsync(responseCacheKey(campaignContactId))
+    }
+  },
+  reloadQuery: async (campaignContactId) => {
+    const questionResponseValues = await r.knex('question_response')
+      .where('question_response.campaign_contact_id', campaignContactId)
+      .select('value', 'interaction_step_id')
+
+    const valueInteractionArray = questionResponseValues.reduce((acc, qrv) => {
+      acc.push(qrv.interaction_step_id, qrv.value)
+      return acc
+    }, [])
+
+    if (r.redis) {
```

Comment:
This condition should go up to line 33 since if there's no cache, we don't need to do the db queries either.